test/api.js

```diff
@@ -0,0 +1,141 @@
+chai.use(dirtyChai)
+
+const multiaddr = require('multiaddr')
+const os = require('os')
+const path = require('path')
+
+const isNode = require('detect-node')
+const isWindows = os.platform() === 'win32'
+
+module.exports = (df, type) => {
+  return () => {
+    const API_PORT = type === 'js' ? '5002' : '5001'
+    const GW_PORT = type === 'js' ? '9090' : '8080'
+
+    const config = {
+      Addresses: {
+        API: `/ip4/127.0.0.1/tcp/${API_PORT}`,
+        Gateway: `/ip4/127.0.0.1/tcp/${GW_PORT}`
+      }
+    }
+
+    describe('ipfs-api version', () => {
+      let ipfsd
+      let api
+
+      before(function (done) {
+        this.timeout(50 * 1000)
+        df.spawn({ start: false, config }, (err, daemon) => {
+          expect(err).to.not.exist()
+          ipfsd = daemon
+          ipfsd.start((err, res) => {
+            expect(err).to.not.exist()
+            api = res
+            done()
+          })
+        })
+      })
+
+      after((done) => ipfsd.stop(done))
+
+      // skip on windows for now
+      // https://github.com/ipfs/js-ipfsd-ctl/pull/155#issuecomment-326970190
+      // fails on windows see https://github.com/ipfs/js-ipfs-api/issues/408
+      if (isWindows || !isNode) {
+        return it.skip('uses the correct ipfs-api')
+      }
+
+      it('uses the correct ipfs-api', (done) => {
+        api.util.addFromFs(path.join(__dirname, 'fixtures/'), {
+          recursive: true
+        }, (err, res) => {
+          expect(err).to.not.exist()
+
+          const added = res[res.length - 1]
+
+          // Temporary: Need to see what is going on on windows
+          expect(res).to.deep.equal([
+            {
+              path: 'fixtures/test.txt',
+              hash: 'Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD',
+              size: 19
+            },
+            {
+              path: 'fixtures',
+              hash: 'QmXkiTdnfRJjiQREtF5dWf2X4V9awNHQSn9YGofwVY4qUU',
+              size: 73
+            }
+          ])
+
+          expect(res.length).to.equal(2)
+          expect(added).to.have.property('path', 'fixtures')
+          expect(added).to.have.property(
+            'hash',
+            'QmXkiTdnfRJjiQREtF5dWf2X4V9awNHQSn9YGofwVY4qUU'
+          )
+          expect(res[0]).to.have.property('path', 'fixtures/test.txt')
+          expect(res[0]).to.have.property(
+            'hash',
+            'Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD'
+          )
+          done()
+        })
+      })
+    })
+
+    describe('validate api', () => {
+      it('starts the daemon and returns valid API and gateway addresses', function (done) {
+        this.timeout(50 * 1000)
+        df.spawn({ config }, (err, res) => {
+          expect(err).to.not.exist()
+          const ipfsd = res
+
+          // Check for props in daemon
+          expect(ipfsd).to.have.property('apiAddr')
+          expect(ipfsd).to.have.property('gatewayAddr')
+          expect(ipfsd.apiAddr).to.not.equal(null)
```

Comment:
FYI: Could be written as `expect(ipfsd.apiAddr).to.not.be.null()`. It's the same for `...to.be.true()`. I won't repeat that.